velox/expression/CoalesceExpr.cpp

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 #include "velox/expression/CoalesceExpr.h"
-#include "velox/expression/VarSetter.h"
+#include "velox/expression/ScopedVarSetter.h"
```

Comment:
ScopedVarSetter is a better name then VarSetter, but I wonder if we could go a step further and introduce a class specifically for setting final selection fields. Currently, the code to do so is quite verbose.

Reply:
When setting final selection, we need to set 2 context fields: isFinalSelection and finalSelection. Perhaps, we could introduce a ScopedFinalSelection class that takes context as a constructor argument as well as the value of the new final selection. The implementation of the class would check if context.isFinalSelection is true and, if so, change it to false and update finalSelection.

Reply:
Sounds good, will implement it as you suggested above.
Note: there will be an exception where we need to use VarSetter like in #2389 where we need to update finalSelection regardless of what isFinalSelection is set to but that should be ok.

Reply:
> Note: there will be an exception where we need to use VarSetter like in https://github.com/facebookincubator/velox/pull/2389

Can we not provide a constructor for that case?

Reply:
Was thinking it might be an overkill for a single use case but I don't have a strong preference either way so I'll add a constructor for that too

Reply:
done